File: Professional/CustomDataProvider/VB.NET/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.vb
Imports GrapeCity.ActiveReports.Extensibility.Data
' <summary>
' Provides the implementation of <see cref="Extensibility.Data.IDataReader"/> for the .NET Framework CSV Data Provider.
' </summary>
Public Class CsvDataProviderFactory
	Inherits DataProviderFactory
	Public Overrides Function CreateCommand() As IDbCommand
		' <summary>
		' Returns a new instance of the <see cref="CsvCommand"/>.
		Return New CsvCommand()
	End Function
	Public Overrides Function CreateConnection() As IDbConnection
		' <summary>
		' Returns a new instance of the <see cref="CsvConnection"/>.
		Return New CsvConnection()
	End Function
End Class

File: Professional/CustomDataProvider/C#/CustomDataProvider/CSVDataProvider/CsvDataProviderFactory.cs
using GrapeCity.ActiveReports.Samples.CustomDataProvider.CsvDataProvider;
using GrapeCity.ActiveReports.Extensibility.Data;
namespace GrapeCity.ActiveReports.Samples.CustomDataProvider.CSVDataProvider
{
	/// <summary>
	/// Implements <see cref="DataProviderFactory"/> for .NET Framework CSV Data Provider.
	/// </summary>
	public class CsvDataProviderFactory : DataProviderFactory
	{
		/// <summary>
		/// Creates a new instance of the <see cref="CsvDataProviderFactory"/> class.
		/// </summary>
		public CsvDataProviderFactory()
		{
		}
		/// <summary>
		/// Returns a new instance of the <see cref="CsvCommand"/>.
		/// </summary>
		public override IDbCommand CreateCommand()
		{
			return new CsvCommand();
		}
		/// <summary>
		/// Returns a new instance of the <see cref="CsvConnection"/>.
		/// </summary>
		public override IDbConnection CreateConnection()
		{
			return new CsvConnection();
		}
	}
}
